src/lib.rs

```diff
@@ -203,11 +203,11 @@ fn encode_str(buf: &mut String, s: &str) {
                 b'\n' => push_escape(buf, 'n'),
                 b'\r' => push_escape(buf, 'r'),
                 b'\t' => push_escape(buf, 't'),
-                0..=0x1F | 0x7F..=0x9F => {
```

Comment:
Wait, I think this whole match is wrong, no? In `c as u8;` we are taking `c` modulo 256, and we don't want `256 + b'\n'` to get treated as `\n`. 

Let's do `let b = u8::try_from(c as u32).unwrap_or(255); // 255 to hit fall-through`, I think that should do the trick? Or maybe just explicitly branch on whether `c` fits in a byte

Reply:
Yup, `256 + b'\n'` (the character `Ċ`) was broken - I added the explicit branch